### PR TITLE
8293064: Remove unused NET_xxx functions

### PR DESCRIPTION
**Pull request by DarraghClarke**
No tests were affected so this is purely a removal PR apart from updating copyright headers.
I ran tests for Tier 1,2&3 and everything seems to be passing

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293064](https://bugs.openjdk.org/browse/JDK-8293064): Remove unused NET_xxx functions

### Reviewers
 * [Chris Hegarty](https://openjdk.org/census#chegar) (@ChrisHegarty - **Reviewer**) ⚠️ Review applies to [caa215ff](https://git.openjdk.org/jdk/pull/10343/files/caa215ff4f71ae3d2d1eb59889e62ee8dfe7e48c)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - Committer)
 * [Aleksei Efimov](https://openjdk.org/census#aefimov) (@AlekseiEfimov - Committer)
 * [Vyom Tewari](https://openjdk.org/census#vtewari) (@vyommani - Committer) ⚠️ Review applies to [caa215ff](https://git.openjdk.org/jdk/pull/10343/files/caa215ff4f71ae3d2d1eb59889e62ee8dfe7e48c)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10343/head:pull/10343` \
`$ git checkout pull/10343`

Update a local copy of the PR: \
`$ git checkout pull/10343` \
`$ git pull https://git.openjdk.org/jdk pull/10343/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10343`

View PR using the GUI difftool: \
`$ git pr show -t 10343`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10343.diff">https://git.openjdk.org/jdk/pull/10343.diff</a>

</details>

### CODE REVIEW

**Review by ChrisHegarty — APPROVED** [review]
LGTM. Great to see that we're at a point where this kinda cleanup can now happen.

File: src/java.base/windows/native/libnet/net_util_md.h [hunk 1]
```diff
@@ -118,10 +116,6 @@ struct ipv6bind {
          (IN6_IS_ADDR_V4MAPPED_LOOPBACK(&(x)->sa6.sin6_addr))) \
 )
 
-JNIEXPORT int JNICALL NET_SocketClose(int fd);
-
-JNIEXPORT int JNICALL NET_Timeout(int fd, long timeout);
```
Comment (djelinski) [inline src/java.base/windows/native/libnet/net_util_md.h:123]: also remove NET_Timeout2 and NET_BindV6

**Review by djelinski — CHANGES_REQUESTED** [review]

**Review by AlekseiEfimov — CHANGES_REQUESTED** [review]
The changes looks good to me with few additional changes needed on Linux platform: 
- `NET_SetTrafficClass`: It was discovered by invoking linker with `--print-gc-sections` option on a Linux-x64 platform. Header file: `net_util_md.h` Source file: `net_util_md.c`, both located in `src/java.base/unix/native/libnet`.
- `NET_ThrowCurrent` on Linux (as already spotted by Daniel). It needs to be removed from `src/java.base/[unix|windows]/native/libnet/net_util_md.c`

**Comment by AlanBateman** [comment]
@AlekseiEfimov InetAddress.ping is not asynchronously closable so I assume there is no need for the remaining NET_* functions to do sys calls that are wrapped with startOp/endOp. Just mentioning it because a future PR could remove a lot more, including the *_close.c files.

**Review by vyommani — APPROVED** [review]
Looks OK to me, i build/run this patch  at my Linux env and worked as expected.

**Comment by Michael-Mc-Mahon** [comment]
Have the AIX changes been verified with an AIX build?

**Comment by DarraghClarke** [comment]
Thanks for the feedback everyone, I made a new commit addressing the comments, let me know if there is anything else

**Comment by DarraghClarke** [comment]
> Have the AIX changes been verified with an AIX build?
Good question, I wasn't able to test those though I tested Mac/Linux/Windows,
@backwaterred Would you be able to test the AIX changes please?

**Review by AlekseiEfimov — APPROVED** [review]
Thanks for cleaning up additional methods, Darragh. 

The following `NET_` methods were removed and no longer referenced from the native code:
`NET_BindV6` `NET_Dup2` `NET_GetFileDescriptorID` `NET_NonBlockingRead` `NET_Read` `NET_RecvFrom` `NET_Send` `NET_SendTo` `NET_SetTrafficClass` `NET_SocketClose` `NET_ThrowCurrent` `NET_ThrowNew` `NET_Timeout` `NET_Timeout2`.

That is a great cleanup :+1:

**Comment by backwaterred** [comment]
I see you ran tests in T1, T2, & T3. I can do the same on AIX. It will take some time to complete, and I'll update when the results are in.

**Review by djelinski — APPROVED** [review]

**Comment by DarraghClarke** [comment]
> I see you ran tests in T1, T2, & T3. I can do the same on AIX. It will take some time to complete, and I'll update when the results are in.
> 
> I found some failures in T2 & T3. This is to be expected as we are less active in adding known failures to the problem lists on AIX. Running again with the master branch to compare, and will update again to let you know the result.
> 
> @DarraghClarke the test results on AIX are looking good. All failures that exist on this branch are present in master as well.

Thanks for testing that!

**Pull request closed without merging**